### DISCUSSION TITLE
How to implement with route filter

### DISCUSSION THREAD

**Discussion by jadsy2107** [codeigniter4/shield#250] [Q&A]
I've been trying to achieve this but cannot seem to ... 
Something like; 
```php
$routes->group('/admin', ['filter' => 'auth'], static function ($routes) {
    $routes->get('/', 'Admin::view');
    $routes->get('(:any)', 'Admin::view/$1');
});
```

And in Auth.php filter:
```php
    public function before(RequestInterface $request, $arguments = null)
    {
       if (!session()->get('isLoggedIn')) {
            return redirect()->to(site_url('login'));
        }
    }
```

**Comment by kenjis** [comment]
If you need to implement your own filter, see the existing filter.

See 
- https://github.com/codeigniter4/shield/blob/develop/docs/install.md#controller-filters
- https://github.com/codeigniter4/shield/blob/develop/src/Filters/SessionAuth.php

**Reply by jadsy2107** [reply]
The controller filters doc has dissappeared 
https://github.com/codeigniter4/shield/blob/develop/docs/install.md#controller-filters

**Reply by datamweb** [reply]
Thanks🥀, edited.

**Comment by MGatner** [comment]
As @kenjis noted this library already comes with filters, and it looks like what you are trying to do just duplicates the existing Session filter. Change your routes to use the library's package and you should be good:
```diff
- $routes->group('/admin', ['filter' => 'auth'], static function ($routes) {
+ $routes->group('/admin', ['filter' => 'session'], static function ($routes) {
```

**Reply by jadsy2107** [reply]
> 

All good I've created my own filter which checks which group and redirects appropriately on login. 
Unless you can modify the session filter?

**Comment by lonnieezell** [comment]
No need to modify the filter to adjust where you go during login - you can use the [loginRedirect](https://github.com/codeigniter4/shield/blob/develop/src/Config/Auth.php#L346) method in the `Auth` config file to check the user and redirect appropriately based on group they belong to. Unfortunately, it's currently buried at the bottom of the config file so it's easy to miss.
